sefaria/helper/crm/salesforce.py

```diff
@@ -145,3 +159,14 @@ def subscribe_to_lists(self, email, first_name=None, last_name=None, lang="en",
         except:
             return False
         return res
+
+    def get_available_lists(self) -> List[str]:
+        try:
+            resource_prefix = f"services/data/v{self.version}/query"
+            endpoint = f"{sls.SALESFORCE_BASE_URL}/{resource_prefix}/"
+            response = self.get(endpoint + "?q=SELECT+Subscriptions__c+FROM+AC_to_SF_List_Mapping__mdt")
+            records = response.json()["records"]
+            return [record["Subscriptions__c"] for record in records]
+        except:
+            print("Unable to retrieve newsletter mailing lists from Salesforce CRM")
```

Comment:
It's not helpful to catch this error here. better to throw a custom error class that can be caught at the API level. There you can return an appropriate error code response (probably 500) and an error message in the API response.

Reply:
Agreed 😃